.github/workflows/publish-docs.yaml

```diff
@@ -0,0 +1,22 @@
+name: Publish Documentation
+on:
+  push:
+    branches: [ documentation/S3CSI-17-setup-github-pages ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: "pip"
+
+      - name: Install locked deps
+        run: pip install -r requirements.txt
+
+      - name: Deploy site to gh-pages
+        run: |
+          mkdocs gh-deploy --clean --force
```

Comment:
That's kind of an old way of deploying on github pages. Would recommend having a look [this workflow](https://github.com/scality/devdocs/blob/main/.github/workflows/publish-docs.yaml) for inspiration if you want to have a look at the new method.

Would recommend this only if you have the time to look at it. The differences between the two methods are pretty minor. (And I can also make a PR for you with the changes if you want)

Reply:
Thanks changed this, but need to review the whole PR again, as things changed I forced pushed everything